modules/setting/config_provider.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package setting
+
+import (
+	"code.gitea.io/gitea/modules/log"
+
+	ini "gopkg.in/ini.v1"
+)
+
+// ConfigProvider represents a config provider
+type ConfigProvider interface {
+	Section(section string) *ini.Section
```

Comment:
In a follow-up PR, we should probably also extract the `*ini.Section`s into their own interfaces to allow for settings of other types.
But that's not for now.

Reply:
The interfaces could be moved into a sub-package and we can have more implementations than just reading file.